### PR TITLE
Add filtering capabilities to SVG selector widget

### PR DESCRIPTION
**Pull request by 3nids**
https://user-images.githubusercontent.com/127259/111769584-af7d3100-88a9-11eb-9b6b-c764d51cdbd4.mov

### CODE REVIEW

**Comment by nyalldawson** [comment]
Nice!

The widget layout looks a bit odd to me -- could you make it extend the whole horizontal size or place it inside the right hand grid cell so that things align nicely?

File: src/gui/symbology/qgssvgselectorwidget.cpp [hunk 1]
```diff
@@ -216,19 +217,25 @@ void QgsSvgGroupLoader::loadGroup( const QString &parentPath )
...
 QgsSvgSelectorListModel::QgsSvgSelectorListModel( QObject *parent, int iconSize )
-  : QAbstractListModel( parent )
-  , mSvgLoader( new QgsSvgSelectorLoader( this ) )
-  , mIconSize( iconSize )
-{
-  mSvgLoader->setPath( QString() );
-  connect( mSvgLoader, &QgsSvgSelectorLoader::foundSvgs, this, &QgsSvgSelectorListModel::addSvgs );
-  mSvgLoader->start();
-}
+  : QgsSvgSelectorListModel( parent, QString(), iconSize )
```
Comment (nyalldawson) [inline src/gui/symbology/qgssvgselectorwidget.cpp:237]: :+1:  nice coding style

File: src/gui/symbology/qgssvgselectorwidget.cpp [hunk 2]
```diff
@@ -395,6 +402,18 @@ QgsSvgSelectorWidget::QgsSvgSelectorWidget( QWidget *parent )
   mGroupsTreeView->setHeaderHidden( true );
   populateList();
 
+  connect( mSvgFilterLineEdit, &QgsFilterLineEdit::textChanged, this, [ = ]( const QString & filterText )
+  {
+    if ( mImagesListView->selectionModel()->selectedIndexes().count() > 0 )
```
Comment (nyalldawson) [inline src/gui/symbology/qgssvgselectorwidget.cpp:407]:
```suggestion
    if ( !mImagesListView->selectionModel()->selectedIndexes().empty() )
```

**Comment by 3nids** [comment]
image

@nyalldawson what do you mean? Shall I increase the spacer size?

**Comment by nyalldawson** [comment]
@3nids 
I mean either this:

image

Or

image

**Comment by 3nids** [comment]
ok, done:
image

**Pull request merged**